src/service/ampdoc-impl.js

```diff
@@ -223,6 +285,30 @@ export class AmpDocSingle extends AmpDoc {
   getUrl() {
     return this.win.location.href;
   }
+
+  /** @override */
+  getBody() {
+    return this.win.document.body;
+  }
+
+  /** @override */
+  onBody(callback) {
+    if (this.win.document.body) {
+      callback(this.win.document.body);
```

Comment:
Yeah, so this is a legacy coming from the original APIs that I proxy here: `waitForBody` and `onDocumentReady` - they are sync/async. It's unfortunate, but also a relatively rare case since both are available in 99% of the cases (especially body-available case). I was going to do a refactoring pass with a simple find/replace-all. Not sure it'd be a good time to address this during the refactoring... WDYT?